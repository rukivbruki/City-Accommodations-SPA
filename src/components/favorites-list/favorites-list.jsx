import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import {compose} from "recompose";

import {ActionCreator} from "../../reducer/data/data";
import CardPlace from "../place-card/place-card.jsx";

class FavoritesList extends PureComponent {
  constructor(props) {
    super(props);

    this._handelClickCity = this._handelClickCity.bind(this);
  }

  _handelClickCity(evt) {
    evt.preventDefault();
    const {onChangeCity, history} = this.props;
    const city = evt.currentTarget.getAttribute(`data-city`);

    onChangeCity(city);
    history.push(city);
  }

  render() {
    const {favorites, cities} = this.props;

    return (
      <ul className="favorites__list" aria-live="polite" role="status">
        {
          cities.map((group) => {
            return <li key={group} className="favorites__locations-items">
              <div className="favorites__locations locations locations--current">
                <div className="locations__item">
                  <button
                    type="button"
                    onClick={this._handelClickCity}
                    data-city={group}
                    className="locations__item-link"
                  >
                    <span>{group}</span>
                  </button>
                </div>
              </div>
              <div className="favorites__places" aria-live="polite" role="status">
                {favorites[group].map((offer) => {
                  return <CardPlace
                    onCardHover={() => {}}
                    key={offer.id}
                    offer={offer}
                  />;
                })}
              </div>
            </li>;
          })
        }
      </ul>
    );
  }
}

FavoritesList.propTypes = {
  favorites: PropTypes.object.isRequired,
  cities: PropTypes.array.isRequired,
  history: PropTypes.object.isRequired,
  onChangeCity: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onChangeCity: (city) => dispatch(ActionCreator.changeCity(city)),
});

const enhance = compose(
    withRouter,
    connect(null, mapDispatchToProps)
);

export {FavoritesList};
export default enhance(FavoritesList);
