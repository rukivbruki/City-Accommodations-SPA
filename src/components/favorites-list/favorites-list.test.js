import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {FavoritesList} from "./favorites-list";

configure({adapter: new Adapter()});

it(`FavoritesList is renders correctly`, () => {
  const favorites = shallow(
      <FavoritesList
        history={{}}
        onChangeCity={jest.fn()}
        cities={[`Amsterdam`]}
        favorites={{'Amsterdam': []}}
      />
  );
  expect(favorites).toMatchSnapshot();
});
