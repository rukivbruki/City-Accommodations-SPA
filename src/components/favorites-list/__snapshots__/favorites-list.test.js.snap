// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList is renders correctly 1`] = `
<ul
  aria-live="polite"
  className="favorites__list"
  role="status"
>
  <li
    className="favorites__locations-items"
    key="Amsterdam"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <button
          className="locations__item-link"
          data-city="Amsterdam"
          onClick={[Function]}
          type="button"
        >
          <span>
            Amsterdam
          </span>
        </button>
      </div>
    </div>
    <div
      aria-live="polite"
      className="favorites__places"
      role="status"
    />
  </li>
</ul>
`;
